src/drivers/driver_zephyr.c

```diff
@@ -1102,6 +1181,121 @@ static void wpa_drv_zep_deinit(void *priv)
 	os_free(if_ctx);
 }
 
+#ifdef CONFIG_WIFI_NM_HOSTAPD_AP
+static void wpa_drv_zep_event_acs_channel_selected(struct zep_drv_if_ctx *if_ctx, union wpa_event_data *event)
+{
+	if (if_ctx->hapd)
+		hostapd_event_wrapper(if_ctx->hapd, EVENT_ACS_CHANNEL_SELECTED, event);
+	else
+		wpa_supplicant_event_wrapper(if_ctx->supp_if_ctx, EVENT_ACS_CHANNEL_SELECTED, event);
+}
+
+static void *wpa_drv_zep_hapd_init(struct hostapd_data *hapd, struct wpa_init_params *params)
+{
+	struct zep_drv_if_ctx *if_ctx              = NULL;
+	const struct zep_wpa_supp_dev_ops *dev_ops = NULL;
+	const struct device *device                = NULL;
+	const struct net_linkaddr *link_addr       = NULL;
+	struct zep_hostapd_dev_callbk_fns callbk_fns;
+
+	device = net_if_get_device(net_if_get_wifi_uap());
+	if (!device) {
+		wpa_printf(MSG_ERROR, "%s: Interface %s not found", __func__, params->ifname);
```

Comment:
This looks weird, shouldn't we try to figure out the device or net_if from the `params->ifname`?
We can get the net_if using `net_if_get_wifi_uap()`, but shouldn't we then also verify that the `params->ifname` is the same as what the net_if is? 
For example something like this:
```
if (net_if_get_wifi_uap() != net_if_get_by_index(net_if_get_by_name(params->ifname))) {
...interface not found...
}
```

Reply:
Updated.